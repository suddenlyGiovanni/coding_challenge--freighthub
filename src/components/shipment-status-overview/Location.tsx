import * as React from 'react'
import styled from '@emotion/styled/macro'

import Place from '@material-ui/icons/LocationCity'
import Typography from '@material-ui/core/Typography'

const LocationContainer = styled.div`
  position: relative;
  display: block;
  /* background-color: papayawhip; */
`
const PlaceName = styled(Typography)`
  position: absolute;
  bottom: -23px;
`

interface Props {
  name: string
  active: boolean
  alignment: 'left' | 'right'
}

export const Location: React.FC<Props> = ({ name, active, alignment }) => {
  return (
    <LocationContainer>
      <Place color={active ? 'primary' : 'inherit'} />
      <PlaceName
        variant="body2"
        noWrap={true}
        style={{
          [alignment]: 0,
        }}
      >
        {name}
      </PlaceName>
    </LocationContainer>
  )
}
