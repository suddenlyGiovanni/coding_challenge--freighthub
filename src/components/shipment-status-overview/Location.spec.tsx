import * as React from 'react'
import { render } from '@testing-library/react'

import { Location, Props } from './Location'

describe('Location', () => {
  test('snapshot', () => {
    // Arrange
    const props: Props = {
      name: 'name',
      active: true,
      alignment: 'left',
    }
    const { container } = render(<Location {...props} />)

    // Assert
    expect(container).toMatchSnapshot()
  })
})
