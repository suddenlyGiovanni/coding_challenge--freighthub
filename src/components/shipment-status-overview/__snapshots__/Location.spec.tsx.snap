// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location snapshot 1`] = `
<div>
  <div
    class="css-178mkk4-LocationContainer e17eqbe40"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <p
      class="MuiTypography-root css-i8g5x1-PlaceName e17eqbe41 MuiTypography-body2 MuiTypography-noWrap"
      data-testid="place-name"
      style="left: 0px;"
    >
      name
    </p>
  </div>
</div>
`;
